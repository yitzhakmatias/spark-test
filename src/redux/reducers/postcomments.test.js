import {
    SET_LOADING,
    GET_COMMENTS,
    CREATE_COMMENT,
    GET_POSTS
} from '../actions/post-comments-actions';
import reducer from './post-comments-reducers';

describe('post comments reducer', () => {
    test('SET_LOADING', () => {
        const action = { type: SET_LOADING };
        expect(reducer(undefined, action)).toMatchSnapshot();
    });
    test('GET_POSTS', () => {
        const action = { type: GET_POSTS };
        expect(reducer(undefined, action)).toMatchSnapshot();
    });
    test('GET_COMMENTS', () => {
        const action = { type: GET_COMMENTS };
        expect(reducer(undefined, action)).toMatchSnapshot();
    });
    test('CREATE_COMMENT', () => {
        const action = {
            type: CREATE_COMMENT,
            payload: {
                postId: 2,
                name: 'Czechmademan',
                body: 'testsetsetset'
                 },
        };
        expect(reducer(undefined, action)).toMatchSnapshot();
    });


});
