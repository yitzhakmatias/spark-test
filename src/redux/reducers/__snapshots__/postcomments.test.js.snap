// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post comments reducer CREATE_COMMENT 1`] = `
Object {
  "comments": Array [
    Object {
      "body": "testsetsetset",
      "name": "Czechmademan",
      "postId": 2,
    },
  ],
  "loading": false,
  "posts": Array [],
  "title": "",
}
`;

exports[`post comments reducer GET_COMMENTS 1`] = `
Object {
  "comments": undefined,
  "loading": false,
  "posts": Array [],
  "title": "",
}
`;

exports[`post comments reducer GET_POSTS 1`] = `
Object {
  "comments": Array [],
  "loading": false,
  "posts": undefined,
  "title": "",
}
`;

exports[`post comments reducer SET_LOADING 1`] = `
Object {
  "comments": Array [],
  "loading": true,
  "posts": Array [],
  "title": "",
}
`;

exports[`user reducer CREATE_COMMENT 1`] = `
Object {
  "comments": Array [
    Object {
      "body": "testsetsetset",
      "name": "Czechmademan",
      "postId": 2,
    },
  ],
  "loading": false,
  "posts": Array [],
  "title": "",
}
`;

exports[`user reducer GET_COMMENTS 1`] = `
Object {
  "comments": undefined,
  "loading": false,
  "posts": Array [],
  "title": "",
}
`;

exports[`user reducer GET_POSTS 1`] = `
Object {
  "comments": Array [],
  "loading": false,
  "posts": undefined,
  "title": "",
}
`;

exports[`user reducer SET_LOADING 1`] = `
Object {
  "comments": Array [],
  "loading": true,
  "posts": Array [],
  "title": "",
}
`;
